subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputHistoryLossIntegrationTest.groovy

```diff
@@ -235,6 +235,9 @@ class StaleOutputHistoryLossIntegrationTest extends AbstractIntegrationSpec {
         javaProjects.each {
             forceDelete(it.redundantSourceFile)
         }
+        if (expectDeprecationWarning) {
```

Comment:
Maybe we can write `if(arguments.contains('--configure-on-demand')){...}`?

Reply:
I didn't want to repeat myself here, so I made it part of the `where:` table instead.